lib/livebook/intellisense.ex

```diff
@@ -538,6 +549,12 @@ defmodule Livebook.Intellisense do
     end
   end
 
+  defp docs_link_hash(:erlang, {:function, function, arity}), do: "##{function}-#{arity}"
+  defp docs_link_hash(:erlang, {:type, type, _arity}), do: "#type-#{type}"
+  defp docs_link_hash(:hexdocs, {:function, function, arity}), do: "##{function}/#{arity}"
+  defp docs_link_hash(:hexdocs, {:type, type, arity}), do: "#t:#{type}/#{arity}"
```

Comment:
Since we already have a separate code branches for `:erlang` and `:hexdocs`, I would just inline this as `case` statements in the respective places, wdyt?

Reply:
I totally see your point, not sure I find this more readable, though (or maybe I misunderstood your suggestion?):

```elixir
cond do
  is_otp? ->
    hash =
      case function_or_type do
        {:function, function, arity} -> "##{function}-#{arity}"
        {:type, type, _arity} -> "#type-#{type}"
        nil -> ""
      end

    url = "https://www.erlang.org/doc/man/#{module_name}.html#{hash}"
    "[View on Erlang Docs](#{url})"

  vsn = app && Application.spec(app, :vsn) ->
    hash =
      case function_or_type do
        {:function, function, arity} -> "##{function}/#{arity}"
        {:type, type, arity} -> "#t:#{type}/#{arity}"
        nil -> ""
      end

    url = "https://hexdocs.pm/#{app}/#{vsn}/#{module_name}.html#{hash}"
    "[View on Hexdocs](#{url})"

  true ->
    nil
end
```

Reply:
It's fine, but alternatively we can have two functions like `erlang_docs_hash` and `hexdocs_hash`, which is closer to your original version?

Reply:
Yeah, if that's fine with you, think I'd like that better. :)

Reply:
Mh. Staring at it for a minute... you're right, having it close to where it's used is better. Think that checking twice for `nil` was what felt off, but it's alright.